src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -123,11 +123,33 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
                 IList<Exception> exceptions = new List<Exception>();
                 if (_hostedServices != null) // Started?
                 {
+                    Queue<Task> tasks = new Queue<Task>();
                     foreach (IHostedService hostedService in _hostedServices.Reverse())
+                    {
+                        switch (_options.BackgroundServiceStopBehavior)
+                        {
+                            case BackgroundServiceStopBehavior.Sequential:
+                                try
```

Comment:
I thought about just creating a `Func<Task, Task>` for handling tasks here instead of duplicating the try/catch logic, or even doing something where you add the task to the queue without starting it, but i felt like this is more easily understandable for future maintenance.